src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,282 @@
+            rules: gpii.devpmt.booleanBinderRules
+        },
+        valueInput: {
+            selector: "valueInput",
+            path: "current.value",
+            rules: {
+                domToModel: {
+                    "": {
+                        transform: {
+                            type: "gpii.devpmt.prefSettingAdjuster.binderToValueInput",
+                            inputPath: "",
+                            schema: "{that}"
+                        }
+                    }
+                },
+                modelToDom: {
+                    "": {
+                        transform: {
+                            type: "gpii.devpmt.prefSettingAdjuster.valueInputToBinder",
+                            inputPath: "",
+                            schema: "{that}"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    templates: {
+        initial: "editprefset-prefSettingAdjuster-widget"
+    },
+    invokers: {
+        saveUpdateValue: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.saveUpdateValue",
+            args: ["{that}", "{editPrefs}"]
+        },
+        cancelEditing: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.cancelEditing",
+            args: ["{that}", "{editPrefs}"]
+        },
+        watchInputKeys: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.watchInputKeys",
+            args: ["{that}", "{arguments}.0"]
+        },
+        updateBlankDisabling: {
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBlankDisabling",
+            args: ["{that}"]
+        }
+    },
+    markupEventBindings: {
+        okButton: {
+            method: "click",
+            args: "{that}.saveUpdateValue"
+        },
+        cancelButton: {
+            method: "click",
+            args: "{that}.cancelEditing"
+        },
+        valueInput: {
+            method: "keypress",
+            args: "{that}.watchInputKeys"
+        }
+    },
+    listeners: {
+        "onMarkupRendered.focus": {
+            funcName: "fluid.focus",
+            args: ["{that}.dom.valueInput"]
+        },
+        "onMarkupRendered.updateBlankDisabling": {
+            func: "{that}.updateBlankDisabling"
+        },
+        "onMarkupRendered.updateBooleanValueLabel": {
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel",
+            args: ["{that}"]
+        }
+    },
+    modelListeners: {
+        "current.blank": {
+            func: "{that}.updateBlankDisabling"
+        },
+        "current.value": [{
+            funcName: "gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel",
+            args: ["{that}"]
+        }]
+    }
+});
+
+gpii.devpmt.prefSettingAdjuster.updateBooleanValueLabel = function (that) {
+    // if (that.model.metadata.schema.type === "boolean") {
+    var booleanValueLabel = that.dom.locate("booleanValueLabel");
+    booleanValueLabel.css("display","inline");
```

Comment:
It's a strange mixture of responsibilities in this function to make some updates unconditionally and to make others in a model-dependent way.

Reply:
This should be mostly simplified now.